api/trace/pass_through_span.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/tag"
+)
+
+type PassThroughSpan struct {
```

Comment:
I'm questioning whether we need this `PassThroughSpan` and `PassThroughTracer`.

My mental model for extracting a context to start a child looks like:

```
  childSpanContext := propagate.Extract(dataFromWire)
  
  ctx, childSpan := tracer.Start("childName", trace.ChildOf(childSpanContext))
```

I guess the key idea here is that we don't need to install the extracted context as the current one in any sense, we only need to pass it to the `Start` call as an option to signify an extracted parent context.

Reply:
> I'm questioning whether we need this `PassThroughSpan` and `PassThroughTracer`.
> 
> My mental model for extracting a context to start a child looks like:
> 
> ```
>   childSpanContext := propagate.Extract(dataFromWire)
>   
>   ctx, childSpan := tracer.Start("childName", trace.ChildOf(childSpanContext))
> ```
> 
> I guess the key idea here is that we don't need to install the extracted context as the current one in any sense, we only need to pass it to the `Start` call as an option to signify an extracted parent context.

Concept for PassThroughTracer was to enable propagating tracecontext without fully enabling tracing. For example, in Istio, envoy-proxy does all the instrumentation. However, it requires[ trace-context header be forwarded ](https://istio.io/docs/tasks/telemetry/distributed-tracing/overview/) through application. In such case PassThroughTracer would be very useful. 

May be it should be pulled out from here for now and can be added later it it is needed.